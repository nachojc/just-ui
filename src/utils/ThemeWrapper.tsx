import { ReactNode } from "react";
import { ThemeProvider } from "styled-components";
import theme from "../styles/themes";

type ThemeName = keyof typeof theme;

type options = {
  themeName: ThemeName;
};
export const wrapWithTheme = (children: ReactNode, options?: options) => {
  const opt: options = { ...{ themeName: "defaultTheme" }, ...options };
  return <ThemeProvider theme={theme[opt.themeName]}>{children}</ThemeProvider>;
};
