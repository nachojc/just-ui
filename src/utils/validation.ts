export const hasText = (value?: string): boolean => {
  return typeof value === "string" && value?.trim().length > 0;
};
