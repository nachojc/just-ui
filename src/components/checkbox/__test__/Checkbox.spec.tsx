import React from "react";
import { render } from "@testing-library/react";
import { Checkbox } from "..";
import { wrapWithTheme } from "../../../utils/ThemeWrapper";

describe("Checkbox", () => {
  it("default render", async () => {
    const { asFragment } = render(wrapWithTheme(<Checkbox />));

    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });

  it("with error", async () => {
    const { asFragment } = render(wrapWithTheme(<Checkbox label="Input" error="no text" />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with display asBlock", async () => {
    const { asFragment } = render(wrapWithTheme(<Checkbox asBlock={true} />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with disabled", async () => {
    const { asFragment } = render(wrapWithTheme(<Checkbox disabled={true} />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("small", async () => {
    const { asFragment } = render(wrapWithTheme(<Checkbox size="small" />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
});
