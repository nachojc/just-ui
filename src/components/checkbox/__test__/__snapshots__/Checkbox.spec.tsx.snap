// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox default render 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-blHHSb zYgxi"
    >
      <input
        class="sc-gtLWhw eXNWMH"
        type="checkbox"
      />
      <span
        class="sc-egkSDF NQFOD"
      />
    </label>
  </div>
</DocumentFragment>
`;

exports[`Checkbox small 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-blHHSb zYgxi"
    >
      <input
        class="sc-gtLWhw eXNWMH"
        type="checkbox"
      />
      <span
        class="sc-egkSDF dMgjqf"
      />
    </label>
  </div>
</DocumentFragment>
`;

exports[`Checkbox with disabled 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-blHHSb giWDzg"
      disabled=""
    >
      <input
        class="sc-gtLWhw eXNWMH"
        disabled=""
        type="checkbox"
      />
      <span
        class="sc-egkSDF cwYloQ"
        disabled=""
      />
    </label>
  </div>
</DocumentFragment>
`;

exports[`Checkbox with display asBlock 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-blHHSb zYgxi"
    >
      <input
        class="sc-gtLWhw eXNWMH"
        type="checkbox"
      />
      <span
        class="sc-egkSDF NQFOD"
      />
    </label>
    <div
      class="sc-fAUdSK CeDQe"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox with error 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-blHHSb zYgxi"
      error="no text"
    >
      Input
      <input
        class="sc-gtLWhw eXNWMH"
        type="checkbox"
      />
      <span
        class="sc-egkSDF lLtdh"
        error="no text"
      />
    </label>
    <div
      class="sc-fAUdSK iJytMv"
      error="no text"
    >
      no text
    </div>
  </div>
</DocumentFragment>
`;
